Goodhosts issues on OSX

We are getting some crashes on OSX.

```
==> default: Issue executing goodhosts CLI: panic: runtime error: slice bounds out of range [381:379]
==> default: 
==> default: goroutine 1 [running]:
==> default: github.com/goodhosts/hostsfile.(*Hosts).removeByPosition(...)
==> default: 	/home/runner/go/pkg/mod/github.com/goodhosts/hostsfile@v0.0.6/hosts.go:297
==> default: github.com/goodhosts/hostsfile.(*Hosts).combineIp(0xc0000d9ac8, 0xc0000145e0, 0x7)
==> default: 	/home/runner/go/pkg/mod/github.com/goodhosts/hostsfile@v0.0.6/hosts.go:285 +0x582
==> default: github.com/goodhosts/hostsfile.(*Hosts).RemoveDuplicateIps(0xc0000d9ac8)
==> default: 	/home/runner/go/pkg/mod/github.com/goodhosts/hostsfile@v0.0.6/hosts.go:224 +0x1f3
==> default: github.com/goodhosts/hostsfile.(*Hosts).Clean(0xc0000d9ac8)
==> default: 	/home/runner/go/pkg/mod/github.com/goodhosts/hostsfile@v0.0.6/hosts.go:130 +0x40
==> default: github.com/goodhosts/cli/cmd.remove(0xc00001ea00, 0x3, 0x4)
==> default: 	/home/runner/work/cli/cli/cmd/remove.go:75 +0x7f3
==> default: github.com/urfave/cli/v2.(*Command).Run(0xc0000127e0, 0xc00001e840, 0x0, 0x0)
==> default: 	/home/runner/go/pkg/mod/github.com/urfave/cli/v2@v2.1.1/command.go:161 +0x4b9
==> default: github.com/urfave/cli/v2.(*App).RunContext(0xc000001200, 0x127d980, 0xc00001a080, 0xc000010050, 0x5, 0x5, 0x0, 0x0)
==> default: 	/home/runner/go/pkg/mod/github.com/urfave/cli/v2@v2.1.1/app.go:302 +0x790
==> default: github.com/urfave/cli/v2.(*App).Run(...)
==> default: 	/home/runner/go/pkg/mod/github.com/urfave/cli/v2@v2.1.1/app.go:211
==> default: main.main()
==> default: 	/home/runner/work/cli/cli/main.go:55 +0x48e
```

Seems the line with the issue is https://github.com/goodhosts/hostsfile/blob/master/hosts.go#L297

I downloaded and ran `sudo goodhosts clean` and got the same error:

```
❯ sudo goodhosts clean
panic: runtime error: slice bounds out of range [362:361]

goroutine 1 [running]:
github.com/goodhosts/hostsfile.(*Hosts).removeByPosition(...)
	/home/runner/go/pkg/mod/github.com/goodhosts/hostsfile@v0.0.6/hosts.go:297
github.com/goodhosts/hostsfile.(*Hosts).combineIp(0xc0000fbc18, 0xc0000c2540, 0x7)
	/home/runner/go/pkg/mod/github.com/goodhosts/hostsfile@v0.0.6/hosts.go:285 +0x582
github.com/goodhosts/hostsfile.(*Hosts).RemoveDuplicateIps(0xc0000fbc18)
	/home/runner/go/pkg/mod/github.com/goodhosts/hostsfile@v0.0.6/hosts.go:224 +0x1f3
github.com/goodhosts/hostsfile.(*Hosts).Clean(0xc0000fbc18)
	/home/runner/go/pkg/mod/github.com/goodhosts/hostsfile@v0.0.6/hosts.go:130 +0x40
github.com/goodhosts/cli/cmd.clean(0xc0000908c0, 0x2, 0x2)
	/home/runner/work/cli/cli/cmd/clean.go:29 +0x7f
github.com/urfave/cli/v2.(*Command).Run(0xc0000aac60, 0xc0000907c0, 0x0, 0x0)
	/home/runner/go/pkg/mod/github.com/urfave/cli/v2@v2.1.1/command.go:161 +0x4b9
github.com/urfave/cli/v2.(*App).RunContext(0xc00008e180, 0x127d980, 0xc000096008, 0xc0000a0020, 0x2, 0x2, 0x0, 0x0)
	/home/runner/go/pkg/mod/github.com/urfave/cli/v2@v2.1.1/app.go:302 +0x790
github.com/urfave/cli/v2.(*App).Run(...)
	/home/runner/go/pkg/mod/github.com/urfave/cli/v2@v2.1.1/app.go:211
main.main()
	/home/runner/work/cli/cli/main.go:55 +0x48e

```

--- a/cmd/main.go
+++ b/cmd/main.go
@@ -17,15 +17,16 @@ func Run(c *cli.Context) error {
 
 func Commands() []*cli.Command {
 	return []*cli.Command{
-		Check(),
-		List(),
 		Add(),
-		Remove(),
-		Debug(),
 		Backup(),
-		Restore(),
-		Edit(),
+		Check(),
 		Clean(),
+		Debug(),
+		Edit(),
+		List(),
+		Remove(),
+		Restore(),
+		Version(),
 	}
 }
 

--- a/cmd/version.go
+++ b/cmd/version.go
@@ -1,0 +1,19 @@
+package cmd
+
+import (
+	"github.com/sirupsen/logrus"
+	"github.com/urfave/cli/v2"
+)
+
+func Version() *cli.Command {
+	return &cli.Command{
+		Name:   "version",
+		Usage:  "",
+		Action: version,
+	}
+}
+
+func version(c *cli.Context) error {
+	logrus.Infof("goodhosts %s", c.Context.Value("version"))
+	return nil
+}

--- a/go.mod
+++ b/go.mod
@@ -3,7 +3,7 @@ module github.com/goodhosts/cli
 go 1.13
 
 require (
-	github.com/goodhosts/hostsfile v0.0.6
+	github.com/goodhosts/hostsfile v0.0.7
 	github.com/olekukonko/tablewriter v0.0.4
 	github.com/sirupsen/logrus v1.4.2
 	github.com/t-tomalak/logrus-easy-formatter v0.0.0-20190827215021-c074f06c5816

--- a/main.go
+++ b/main.go
@@ -1,6 +1,7 @@
 package main
 
 import (
+	"context"
 	"fmt"
 	"io/ioutil"
 	"os"
@@ -12,6 +13,8 @@ import (
 	"github.com/urfave/cli/v2"
 )
 
+var version = "dev"
+
 func main() {
 	app := &cli.App{
 		Name:     "goodhosts",
@@ -19,6 +22,7 @@ func main() {
 		Action:   cmd.DefaultAction,
 		Commands: cmd.Commands(),
 		Before: func(ctx *cli.Context) error {
+			ctx.Context = context.WithValue(ctx.Context, "version", version)
 			if ctx.Bool("debug") {
 				logrus.SetLevel(logrus.DebugLevel)
 			} else {

